software/database/src/main/java/org/apache/brooklyn/entity/database/postgresql/PostgreSqlNode.java

```diff
@@ -90,6 +95,17 @@
             String.class, "postgresql.username", "Username of the database user");
     
     String DEFAULT_USERNAME = "postgresqluser";
+
+    @SetFromFlag("roles")
+    ConfigKey<Map<String, Map>> ROLES = new MapConfigKey(Map.class, "postgresql.roles",
+            "Set roles with properties and permissions. Shoud be a map with keys equal to role names and values a map of the type:" +
+                  "key equal to `properties` and value - the role properties that should be in the query after `WITH` statement" +
+                    "key equal to `privileges` and value - the `GRANT` query value between the `GRANT` and `TO` statements"
+    );
```

Comment:
Maybe worth to add an actual example within the description of what the map should look like?